perfact/zodbsync/object_types.py

```diff
@@ -418,9 +420,11 @@ def write(obj, data):
             connection_string=data['connection_string'],
             zdatetime=data['zdatetime'],
             tilevel=data['tilevel'],
-            autocommit=data['autocommit'],
-            readonlymode=data['readonlymode'],
+            autocommit=data.get('autocommit', False),
+            readonlymode=data.get('readonlymode', False),
             encoding=data['encoding'],
+            use_tpc=data.get('use_tpc', False),
+            datetime_str=data.get('datetime_str', False),
```

Comment:
Will this also work with old versions of `ZPsycopgDA` or versions that are not our fork? I guess they will break because they do not expect the additional arguments.

In that case, maybe we need to only add the new arguments if they are really present in the object (using `hasattr`) when reading and construct a params dictionary here which is passed using `**params` to `manage_edit`.

Reply:
Good point.